src/main/java/wooteco/subway/domain/Age.java

```diff
@@ -0,0 +1,33 @@
+package wooteco.subway.domain;
+
+import java.util.Arrays;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public enum Age {
+
+    INFANTS(age -> age >= 0 && age < 6, fare -> 0),
+    CHILDREN(age -> age >= 6 && age < 13, fare -> (int) ((fare - 350) * (1 - 0.5))),
+    TEENAGERS(age -> age >= 13 && age < 19, fare -> (int) ((fare - 350) * (1 - 0.2))),
+    ADULTS(age -> age >= 19, fare -> fare),
+    ;
+
+    private final Predicate<Integer> ageDiscriminator;
+    private final Function<Integer, Integer> fareCalculator;
```

Comment:
```suggestion
    private final UnaryOperator<Integer> fareCalculator;
```
이런식으로도 함수형 인터페이스를 사용할순 있습니다 :)